File: solutions/typescript/prime-factors/1/prime-factors.ts
export function calculatePrimeFactors(number: number): number[] {
  const factors: number[] = [];
  let divisor = 2;

  while (number > 1) {
    while (number % divisor === 0) {
      factors.push(divisor);
      number = number / divisor;
    }
    divisor++;
  }

  return factors;
}

File: solutions/julia/prime-factors/1/prime-factors.jl
function prime_factors(n)
    factors = Int[]
    divisor = 2
    
    while n > 1
        while n % divisor == 0
            push!(factors, divisor)
            n = div(n, divisor)
        end
        divisor += 1
    end
    
    return factors
end
